README.md

```diff
@@ -1,8 +1,69 @@
-# xlang
+# yarpc/xlang [![Build Status](https://travis-ci.org/yarpc/xlang.svg?branch=master)](https://travis-ci.org/yarpc/xlang)
 
-[![Build Status](https://travis-ci.org/yarpc/xlang.svg?branch=master)](https://travis-ci.org/yarpc/xlang)
+A 2mb Docker appliance for running cross-repo integration tests; xlang is:
+
+* Portable - runs anywhere Docker is installed, eg Travis & locally.
+* General - can be used to test sets of libraries and microservices.
+* Flexible - test all combinations of behaviors using custom matrix dimensions.
+* Decentralized - each repo can configure and run Xlang independently from the others.
+* Light - run Xlang for every commit on every repo in parallel.
+* Easy - run integration tests on a large project without installing every component.
+
+## How It Works
+
+Xlang is [published in Docker Hub](https://hub.docker.com/r/yarpc/xlang/) and is
+meant to be used with [Docker Compose](https://docs.docker.com/compose/) directly from your repos.
+
+Given the following `docker-compose.yml`, Xlang will initiate an integration test for clients
+`alpha` and `omega` for every combination of `behavior` and `speed`:
+
+```yml
+xlang:
+    image: yarpc/xlang
+    links:
+        - alpha
+        - omega
+    environment:
+        - XLANG_CLIENTS=alpha,omega
+        - XLANG_DIMENSION_BEHAVIOR=dance,run
+        - XLANG_DIMENSION_SPEED=fast,slow
+
+alpha: &alpha
```

Comment:
It's YAML dude - I'm storing a reference to `alpha`, then expanding it in `omega`.